### ISSUE TITLE
Use `worker_threads`

### ISSUE THREAD

**Issue by JounQin** [sindresorhus/make-synchronous#9]
See https://github.com/rx-ts/synckit/blob/perf/make-synchronous/benchmarks/benchmark.txt

It's about 300x slower.

**Comment by sindresorhus** [comment]
Yeah, I've been thinking about moving to `worker_threads` (and removing the current child process implementation). When I initially made this package, I was not aware that it could be done with `worker_threads` too.

**Comment by JounQin** [comment]
Can I recommend user's to use `synckit` instead in the README?

**Comment by sindresorhus** [comment]
I plan on supporting `worker_threads` here soon.

**Comment by sindresorhus** [comment]
> It's about 300x slower.

Your benchmark is flawed. You should run a heavy operation in the worker to actually test it, not just a lightweight task. Also, the task you are running in the worker (`fs.promises.readFile`) is already run in a background thread by Node.js.

**Comment by JounQin** [comment]
> I plan on supporting `worker_threads` here soon.

That's great, I would follow your changes then.

**Comment by mmkal** [comment]
@sindresorhus I came here wondering what the advantages are of this over [synckit](https://www.npmjs.com/package/synckit). synckit seems well thought, fairly feature-rich out and [widely used](https://npmtrends.com/make-synchronous-vs-synckit) - do you think it's worth maintaining this rival at this point? Is there something missing from synckit that make-synchronous does better, or will do better?

This is a genuine question - it's often hard to tell what the tradeoffs of various libraries are, and maintainers tend to know best.

**Comment by sindresorhus** [comment]
@mmkal This executes in a subprocess which has some benefits:
- Complete isolation.
- Cannot hang the process (worker threads can potentially hang the main process).
- Separate memory limit, so you could execute something that takes a lot of memory.

Yes, it's slower to launch a subprocess, but that only matters if you use it for short-running tasks. For long-running intensive tasks, the difference is neglible.

**Comment by JounQin** [comment]
> Yes, it's slower to launch a subprocess, but that only matters if you use it for short-running tasks. For long-running intensive tasks, the difference is neglible.

Or if your task needs to be run many times, there will be huge difference too.
